File: database_api/app/routes/admins.py
from flask import Blueprint, request, jsonify
from app.utils.tables.admins import Admins

admins_bp = Blueprint('admins_bp', __name__)

# Insert Data
@admins_bp.route('/', methods=['POST'])
def create_item():
    adm = Admins()
    data = request.get_json()
    
    result = adm.insert(data)       
    adm.close()
    
    if result is False: return jsonify({'message': 'Admin not created'}), 400
    else: return jsonify({'message': 'Admin not created', 'id': result}), 201

# Select Data
@admins_bp.route('/<id>', methods=['GET'])
def select_item(id):
    adm = Admins()
    
    if id == "all": result = adm.select_all()
    else: result = adm.select(id)
    
    adm.close()
    
    if result is False: return jsonify({'message': 'Admin not found'}), 404
    else: return jsonify(result), 200

# Update Data
@admins_bp.route('/<id>', methods=['PUT'])
def update_item(id):
    adm = Admins()
    data = request.get_json()
    
    result = adm.update(id, data)
    adm.close()
    
    if result: return jsonify({'message': 'Admin updated'}), 200
    else: return jsonify({'message': 'Admin not updated'}), 404

# Delete Data
@admins_bp.route('/<id>', methods=['DELETE'])
def delete_item(id):
    adm = Admins()
    
    result = adm.delete(id)
    adm.close()
    
    if result: return jsonify({'message': 'Admin deleted'}), 200
    else: return jsonify({'message': 'Admin not found'}), 404

File: frontend/src/modules/admins.js
const axios = require("axios");
const Admin = require("../class/admin").Admin;

const link = "http://localhost:3030/api/admins/";

async function post(doc) {
    result = (await axios.post(link, doc)).data;

    if (result.message == "Admin created") return result.id;
    else return false;
}

async function get(id){
    result = (await axios.get(link + `${id}`)).data;

    if (result.message == "Admin not found") return false;

    return convert(result);
}

async function get_all(){
    result = (await axios.get(link + "all")).data;

    if (result.message == "Admin not found") return false;

    result.forEach( (val, i, array) => {
        array[i] = convert(val)
    });

    return result;
}

async function put(doc, id) {
    result = (await axios.put(link + `${id}`, doc)).data;
    
    if (result.message == "Admin updated") return true;
    else return false;
}

async function del(id) {
    result = (await axios.delete(link + `${id}`)).data;
    
    if (result.message == "Admin deleted") return true;
    else return false;
}

function convert(doc){
    return new Admin (
        //id
        doc.id,
        //name
        doc.name,
        //email
        doc.email,
        //passwrd
        doc.passwrd,
        //derp
        doc.derp
    )
}

module.exports = {
    "post" : post,
    "get" : get,
    "get_all": get_all,
    "put" : put,
    "del" : del
};